CMakeLists.txt

```diff
@@ -28,6 +28,10 @@ enable_language(CXX)
 #####################################################################
 
 include(CheckCXXCompilerFlag)
+check_cxx_compiler_flag("-std=c++11" COMPILER_SUPPORTS_CXX11)
+if(COMPILER_SUPPORTS_CXX11)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=gnu++0x")
```

Comment:
That is a bit strange. You check for `-std=c++11` and then you inject `-std=gnu++0x` ? And why do you want to add `-Wall`? - That has nothing to do with c++11!